### PR TITLE
Add assembly strong name.

### PR DESCRIPTION
**Pull request by mfilippov**
Add strong name to allow use this library in projects with strong names.

### CODE REVIEW

**Comment by mfilippov** [comment]
@tmds ping.

**Comment by tmds** [comment]
I will make some time for this tomorrow.

**Comment by mfilippov** [comment]
@tmds Thank you!

File: src/Tmds.DBus/CodeGen/DynamicAssembly.cs [hunk 1]
```diff
@@ -24,8 +24,17 @@ internal class DynamicAssembly
 
         private DynamicAssembly()
         {
-            _assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName(Tmds.DBus.Connection.DynamicAssemblyName), AssemblyBuilderAccess.Run);
-            _moduleBuilder = _assemblyBuilder.DefineDynamicModule(Tmds.DBus.Connection.DynamicAssemblyName);
+            var keyStream = typeof(DynamicAssembly).GetTypeInfo().Assembly.GetManifestResourceStream("Tmds.DBus.sign.snk");
```
Comment (tmds) [inline src/Tmds.DBus/CodeGen/DynamicAssembly.cs:27]: Should we dispose the Stream?

File: src/Tmds.DBus/CodeGen/DynamicAssembly.cs [hunk 2]
```diff
@@ -24,8 +24,17 @@ internal class DynamicAssembly
...
+            if (keyStream == null) throw new InvalidOperationException("'Tmds.DBus.sign.snk' not found in resources");
+            var keyBuffer = new byte[keyStream.Length];
```
Comment (tmds) [inline src/Tmds.DBus/CodeGen/DynamicAssembly.cs:29]: Should we hard-code these bytes in order to not have to pack `Tmds.DBus.sign.snk`?
Is this the same data as the hex-encoded `PublicKey` in `DynamicAssemblyName`?

**Comment by tmds** [comment]
@mfilippov Thanks for adding this.  haven't used strong naming in real life. I'm curious to learn a bit more. I guess you need this because you want to strong name your own assembly? Do you have a strong reason to strong name your assembly? Or is is also a requirement that ripples down on you?

You're signing 3 things: Tmds.DBus, Tmds.DBus.Tool, and the dynamic assembly: Tmds.DBus.Emit. Do you need all 3 to be strongly named?

PR looks good, just a few nits.

File: src/Tmds.DBus/CodeGen/DynamicAssembly.cs (hunk 2 above)
Comment (mfilippov) [reply src/Tmds.DBus/CodeGen/DynamicAssembly.cs]: It is possible but looks a little strange. snk in resources add a similar amount of bytes.

**Comment by mfilippov** [comment]
@tmds Without strong name assembly could not be identified by version. And if you have a strong name on the entry point, you can't load assembly without a strong name. We want to use Avalonia in our project, which has a strong-named entry point.

**Comment by mfilippov** [comment]
@tmds Thank you! Could you publish a new version on nuget.org?

**Comment by tmds** [comment]
Thanks @mfilippov! I'll release a new version of Tmds.DBus by next week.

**Comment by tmds** [comment]
@mfilippov it took a week longer, but available now in 0.9.0 on nuget.org.

**Comment by mfilippov** [comment]
@tmds Great thank you!

**Pull request merged**